tools/talis/s3.go

```diff
@@ -145,3 +128,49 @@ func downloadS3Directory(ctx context.Context, client *s3.Client, bucket, prefix,
 
 	return nil
 }
+
+func createS3Client(ctx context.Context, cfg Config) (*s3.Client, error) {
+	s3cfg := cfg.S3Config
+	var awsCfg aws.Config
+	var err error
+	if cfg.S3Config.Endpoint != "" {
+		customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) { //nolint:staticcheck
```

Comment:
using this deprecated interface as it's easier and provides an automatic way to parse any s3 compatible buckets.